### PR TITLE
Fix trying to setState when the component is umounted

### LINKED ISSUE #27: handleInput still receiving data after the component is unmounted
This happens in a corner case when the component is unmounted due to a key input (which makes the component doesn't render anymore) but the event emitter is already iterating the listeners.

The reason this happens is, because the events emitter create a copy of the listeners to iterate them, and the `stdin.removeListener` call happens after the copy is done, therefore the `handleInput` listener receives the event and tries to update the state in a unmounted component, triggering a warning/error

### PR DESCRIPTION
**Pull request by danikaze**
Fixes #27

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -40,6 +40,8 @@ class TextInput extends PureComponent {
 		cursorWidth: 0
 	}
 
+	isMounted = false;
```
Comment (vadimdemedes) [inline src/index.js]: I think you can make it `true` by default and skip setting it to `true` in `componentDidMount()`.

Comment (danikaze) [reply src/index.js]: If you think that's better I'll change it.
I just wanted to make the process clear :)

Comment (vadimdemedes) [reply src/index.js]: Sure, I don't mind it.

**Comment by vadimdemedes** [comment]
Thanks!

**Comment by vadimdemedes** [comment]
Published https://github.com/vadimdemedes/ink-text-input/releases/tag/v3.1.1.

**Pull request merged**